src/bitmap/sorted_u16_vec.rs

```diff
@@ -164,6 +172,53 @@ impl Default for SortedU16Vec {
     }
 }
 
+#[derive(Debug)]
+pub struct Error {
+    index: usize,
+    kind: ErrorKind,
+}
+
+#[derive(Debug)]
+pub enum ErrorKind {
+    Duplicate,
+    OutOfOrder,
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self.kind {
+            ErrorKind::Duplicate => {
+                write!(f, "Duplicate element found at index: {}", self.index)
+            }
+            ErrorKind::OutOfOrder => {
+                write!(f, "An element was out of order at index: {}", self.index)
+            }
+        }
+    }
+}
+
+impl std::error::Error for Error {}
+
+impl TryFrom<Vec<u16>> for SortedU16Vec {
+    type Error = Error;
+
+    fn try_from(value: Vec<u16>) -> Result<Self, Self::Error> {
+        let mut iter = value.iter().enumerate();
+        if let Some((_, mut prev)) = iter.next() {
+            for (i, cur) in iter {
+                match cur.cmp(prev) {
+                    Ordering::Less => return Err(Error { index: i, kind: ErrorKind::OutOfOrder }),
+                    Ordering::Equal => return Err(Error { index: i, kind: ErrorKind::Duplicate }),
+                    Ordering::Greater => {}
```

Comment:
I prefer that we use the unit `()` type instead of an empty scope `{}` which itself return unit `()` 😃

```suggestion
                    Ordering::Greater => (),
```